framework/src/main/java/org/checkerframework/framework/type/AbstractViewpointAdapter.java

```diff
@@ -64,9 +64,7 @@ public void viewpointAdaptMember(
         if (!shouldAdaptMember(memberType, memberElement)) {
             return;
         }
-
-        AnnotatedTypeMirror decltype = atypeFactory.getAnnotatedType(memberElement);
-        AnnotatedTypeMirror combinedType = combineTypeWithType(receiverType, decltype);
+        AnnotatedTypeMirror combinedType = combineTypeWithType(receiverType, memberType);
```

Comment:
replacing `memberType` with `decltype` isn't nice, as `membertype` is a concrete type and already be performed with substitution. `combinetypewithtype` should be the one to perform substition.